src/app/features/create-task/create-task.component.ts

```diff
@@ -25,12 +25,15 @@ export class CreateTaskComponent {
     status: ['', [Validators.required]],
     createdOn: ['', [Validators.required]],
   });
+  private ngUnsubscribe: Subject<void> = new Subject();
 
   constructor(private fb: FormBuilder, private taskService: TaskService) {
   }
 
   onSubmit(): void {
-    this.taskService.addTask(this.currentTask.value).subscribe();
+    this.taskService.addTask(this.currentTask.value)
+      .pipe(takeUntil(this.ngUnsubscribe))
```

Comment:
if you're going to use the same apporach for each unsubscribe, please keep it consistent